.github/actions/gradle-args/action.yml

```diff
@@ -21,25 +21,40 @@ runs :
       run : |
         runnerOS=$RUNNER_OS
 
-        # Set common JVM arguments
-        jvmArgs="-XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
+        totalMemory=0
+        # How much memory does the OS require? 
+        memoryOverhead=0
 
         case $runnerOS in
           macOS)
-            jvmArgs="-Xmx10g -XX:MaxMetaspaceSize=5g $jvmArgs"
+            totalMemory=$(sysctl -n hw.memsize | awk '{print int($1/1024/1024/1024+0.5)}')
+            memoryOverhead=3
             ;;
           Linux)
-            jvmArgs="-Xmx4g -XX:MaxMetaspaceSize=3g $jvmArgs"
+            totalMemory=$(awk '/MemTotal/ {print int($2/1024/1024+0.5)}' /proc/meminfo)
+            memoryOverhead=3
             ;;
           Windows)
-            jvmArgs="-Xmx3g -XX:MaxMetaspaceSize=756m $jvmArgs"
+            # Fetch and parse memory in MB, then convert to GB
```

Comment:
Wow, are you actually set up to test this?

Reply:
ChatGPT was kind enough to show me how to do this.  It does seem to work, but I do not take credit. :)